apis/apiextensions/fn/io/v1alpha1/functionio_types.go

```diff
@@ -227,6 +228,22 @@ type DesiredReadinessCheck struct {
 	// type.
 	// +optional
 	MatchInteger *int64 `json:"matchInteger,omitempty"`
+
+	// MatchCondition specifies the condition you'd like to match if you're using "MatchCondition" type.
+	// +optional
+	MatchCondition *MatchConditionReadinessCheck `json:"matchCondition,omitempty"`
+}
+
+// MatchConditionReadinessCheck is used to indicate how to tell whether a resource is ready
+// for consumption
+type MatchConditionReadinessCheck struct {
+	// Type indicates the type of condition you'd like to use.
+	// +kubebuilder:default="Ready"
+	Type string `json:"type,omitempty"`
+
+	// Status is the status of the condition you'd like to match.
+	// +kubebuilder:default="True"
+	Status string `json:"status,omitempty"`
```

Comment:
These fields aren't optional, right? If so I think we should remove the `omitempty` per https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required (same elsewhere).

Reply:
you are right, these are actually required fields with a default, I removed them everywhere, which correctly resulted in these fields being marked as required in the schema.